tests/integration_tests/performance/vsock_throughput_test_config.py

```diff
@@ -0,0 +1,320 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Configuration file for the VSOCK throughput test."""
+
+IPERF3 = "iperf3-vsock"
+THROUGHPUT = "throughput"
+THROUGHPUT_TOTAL = "total"
+DURATION = "duration"
+DURATION_TOTAL = "total"
+CPU_UTILIZATION_HOST = "cpu_utilization_host"
+CPU_UTILIZATION_GUEST = "cpu_utilization_guest"
+BASE_PORT = 5201
+CPU_UTILIZATION_VMM_TAG = "vmm"
+CPU_UTILIZATION_VCPUS_TOTAL_TAG = "vcpus_total"
+IPERF3_CPU_UTILIZATION_PERCENT_OUT_TAG = "cpu_utilization_percent"
+IPERF3_END_RESULTS_TAG = "end"
+TARGET_TAG = "target"
+DELTA_PERCENTAGE_TAG = "delta_percentage"
+THROUGHPUT_UNIT = "Mbps"
+DURATION_UNIT = "seconds"
+CPU_UTILIZATION_UNIT = "percentage"
+
+
+CONFIG = {
+    "time": 20,  # seconds
+    "server_startup_time": 2,  # seconds,
+    "load_factor": 1,  # nr of iperf clients/vcpu
+    "modes": {
+        "g2h": [""],
+        "h2g": ["-R"],
+        "bd": ["", "-R"]
+    },
+    "protocols": [
+        {
+            "name": "tcp",
+            "omit": 3,  # seconds
+            "payload_length": ["1024K",  None],  # None == Default
+        }
+    ],
+    "hosts": {
+        "instances": {
+            "m5d.metal": {
+                "cpus": [
```

Comment:
Is there a way to auto generate these baselines for each CPU ?

Reply:
There is a sequence of manual steps that one can do, in order to gather baselines. The problem is that this is not yet fully stable. For m5d.metal, we are seeing two types of CPUs, so running a script that launches tests on metal instances cannot deterministically use 50%-50% both CPU types.

Once we define a clear workaround to this issue, we can have clear runbooks that automate this process